File: data_engineer/scala/LoadMovie/src/main/scala/LoadVotos.scala
import org.apache.spark.sql._
import org.apache.log4j._
import org.apache.spark.sql.functions._
import java.util.Properties

object LoadVotos {
  val num = 1
  val path = "/Users/eduardoalberto/LoadFile/data-"+num+".csv"
  val ldir = "/Users/eduardoalberto/LoadFile/"
  // usuario senha
  val connectionProperties = new Properties()
  connectionProperties.put("user", "sa")
  connectionProperties.put("password", "Numsey@Password!")

  def main(args: Array[String]): Unit = {

    // Set the log level to only print errors
    Logger.getLogger("org").setLevel(Level.ERROR)

    val spark = SparkSession
      .builder
      .appName("SparkSQL")
      .master("local")
      .getOrCreate()

    val arq = new java.io.File(path).exists()
    if (arq){
      val df00 = spark.read.option("delimiter", "\t").option("header","true").csv(path)
      val df01 = df00.select(col("tconst"), col("averageRating"),current_date().alias("dt_atualizacao"))
      df01.limit(20000).write.mode(SaveMode.Append).jdbc("jdbc:sqlserver://localhost:1433;databaseName=DBDWP511", "tb_votos", connectionProperties)
      val total = spark.read.jdbc("jdbc:sqlserver://localhost:1433;databaseName=DBDWP511", "tb_votos", connectionProperties).count()
      println("TOTAL DE "+total+" REGISTROS")
    }
    else{
      println("arquivo não encontrado")

    }

  }

}

File: data_engineer/pyspark/LoadMovie/LoadVotos.py
from pyspark.sql import SparkSession
from pyspark.sql.functions import *
from pyspark.sql.types import *
import os 
import os.path as path
import datetime as dt
import sys

#variavel de data / Diretorio dos arquivos
dt_ultima_data = (dt.date.today() - dt.timedelta(2)).strftime('%Y%m%d')
pfile = f"/Users/eduardoalberto/LoadFile/data_{dt_ultima_data}.csv"
ldir  =  "/Users/eduardoalberto/LoadFile/"

# print(dt_ultima_data)

if __name__ == "__main__":
      spark=SparkSession.builder.master("local[1]")\
      .appName("movie")\
      .config("spark.driver.extraClassPath","/Users/eduardoalberto/opt/spark/jars/mssql-jdbc-7.4.1.jre8.jar" ) \
      .getOrCreate()
      
      # verifica se arquivo existe
      if path.exists(pfile):
            df01 = spark.read.option("delimiter", "\t").option("header", True).option("inferSchema", True).csv(pfile)
            df01 = df01.select("tconst","averageRating","numVotes",current_date().alias("dt_atualizacao"))
            df01.select("*").limit(1000).write.mode("overwrite").jdbc("jdbc:sqlserver://localhost:1433;databaseName=DBDWP511", "tb_votos",properties={"user": "sa", "password": "Numsey@Password!"})
            total = spark.read.jdbc("jdbc:sqlserver://localhost:1433;databaseName=DBDWP511", "tb_votos",properties={"user": "sa", "password": "Numsey@Password!"})
            total_registro = total.count()
            print(f"TABELA CRIANDO NO SQL SERVER, TOTAL DE REGISTRO:{total_registro}")
      else:
            obj = os.scandir(ldir)
            print(f"DIRETORIO DOS ARQUIVOS: {ldir}")
            for entry in obj :
                if entry.is_dir() or entry.is_file():
                    print(entry.name)
            obj.close()









